engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/desugar/LambdaShorthandToLambdaMini.scala

```diff
@@ -0,0 +1,380 @@
+package org.enso.compiler.pass.desugar
+
+import org.enso.compiler.context.{CompilerContext, FreshNameSupply}
+import org.enso.compiler.core.{CompilerError, IR}
+import org.enso.compiler.core.ir.expression.{Application, Case, Operator}
+import org.enso.compiler.core.ir.{
+  CallArgument,
+  DefinitionArgument,
+  Expression,
+  Function,
+  IdentifiedLocation,
+  Name,
+  ProcessingPass
+}
+import org.enso.compiler.pass.{IRPass, MiniIRPass}
+
+class LambdaShorthandToLambdaMini(
+  protected val freshNameSupply: FreshNameSupply
+) extends MiniIRPass[ShouldSkipMeta] {
+
+  override def prepare(ir: IR): Unit = {
+    ir match {
+      case Application.Prefix(fn, args, _, _, _) =>
```

Comment:
What's the idea behind this pattern? Do something different (`skip`) for direct children of `Application.Prefix`?

In the [MiniPassIR prepare(IR ir)](https://github.com/enso-org/enso/pull/11191/files#r1778478784) approach one would just return:
```scala
ir match {
  case Application.Prefix => new LambdaShorthandToLambdaMini(freshNameSupply, true)
}
```
assuming there is a constructor `LambdaShorthandToLambdaMini(freshNameSupply: FreshNameSupply, isParentApplicationPrefix: Boolean)`. The (direct) children would then be processed knowing they should _skip meta_.

Reply:
API changed in 985b8c24b2932f3c0f9700a2040bec6ded3364e0. No more metadata attaching.